modules/actions/analytics.js

```diff
@@ -99,13 +118,91 @@ export function getAppGeoDistribution(name, filterId) {
 							appName,
 						},
 					),
-				))
-			.catch(error => dispatch(
+				),
+			)
+			.catch((error) =>
+				dispatch(
 					createAction(
 						AppConstants.APP.ANALYTICS.GET_GEO_DISTRIBUTION_ERROR,
 						null,
 						error,
 					),
-				));
+				),
+			);
+	};
+}
+
+export function getAppAnalyticsInsights(name) {
+	return (dispatch, getState) => {
+		const appName = name || get(getState(), '$getCurrentApp.name', 'default');
+		dispatch(createAction(AppConstants.APP.ANALYTICS.GET_INSIGHTS));
+		return getAnalyticsInsights(appName)
+			.then((res) => {
+				if (res.status === 404) {
```

Comment:
@jyash97 How did you test it?
It'll not work since you're getting `doGet` method in `getAnalyticsInsights` which will reject the promise if there is an error. There is a param in `doGet` that can be used to get the raw response.

Reply:
It is working I tested it, doGet is not rejecting if we get `200` as status. ES gives `200` status on the error as well. I have shown this in video - https://www.loom.com/share/30251bc91d744325aeeb3b3ebdd4e71a

Please check the video.